### PR TITLE
'HA masters 3x all resource'

### PR DESCRIPTION
**Pull request by calvix**
changes:
* add DNS RecordSets for the etcd bound to Eni IP
* rewrite code and templates to have the possibility to create 1 or 3 masters

### CODE REVIEW

File: service/controller/resource/tccpn/create.go [hunk 1]
```diff
@@ -146,11 +147,18 @@ func (r *Resource) createStack(ctx context.Context, cr infrastructurev1alpha2.AW
...
+	}
+
+	masterCount := 1
+
 	var templateBody string
 	{
 		r.logger.LogCtx(ctx, "level", "debug", "message", "computing the template of the tenant cluster's control plane nodes cloud formation stack")
 
-		params, err := newTemplateParams(ctx, cr, r.apiWhitelist, r.route53Enabled)
+		params, err := newTemplateParams(ctx, cr, r.route53Enabled, masterCount, key.ClusterBaseDomain(*clusterCr))
```
Comment (calvix) [inline service/controller/resource/tccpn/create.go:161]: the whitelist was never actually used anywhere so I removed it

File: service/controller/resource/tccpn/create.go [hunk 2]
```diff
@@ -269,32 +291,57 @@ func newENI(ctx context.Context, cr infrastructurev1alpha2.AWSControlPlane) (*te
 	{
 		zones := cc.Spec.TenantCluster.TCCP.AvailabilityZones
 		for _, az := range zones {
-			if az.Name != key.ControlPlaneAvailabilityZones(cr)[0] {
-				continue
-			}
```
Comment (calvix) [inline service/controller/resource/tccpn/create.go:274]: this needs some testing

File: service/controller/resource/tccpn/create_test.go [hunk 3]
```diff
@@ -30,43 +30,51 @@ var update = flag.Bool("update", false, "update .golden CF template file")
...
+			masterCount:    3,
+			baseDomain:     "example.com",
+		},
+		{
+			name:           "case 3: basic test with encrypter backend KMS, route53 disabled, 3 masters",
+			ctx:            unittest.DefaultContextControlPlane(),
+			cr:             unittest.DefaultControlPlane(),
+			route53Enabled: false,
+			masterCount:    3,
+			baseDomain:     "example.com",
```
Comment (calvix) [inline service/controller/resource/tccpn/create_test.go:71]: added test cases for 3 masters

File: service/controller/resource/tccpn/testdata/case-0-basic-test-with-encrypter-backend-KMS-route53-enabled.golden [hunk 4]
```diff
@@ -6,22 +6,23 @@ Outputs:
   OperatorVersion:
```
Comment (calvix) [inline service/controller/resource/tccpn/testdata/case-0-basic-test-with-encrypter-backend-KMS-route53-enabled.golden:6]: to see the changes it's probabbly best to check the golden files diff

File: service/controller/resource/tccpn/testdata/case-0-basic-test-with-encrypter-backend-KMS-route53-enabled.golden [hunk 5]
```diff
@@ -6,22 +6,23 @@ Outputs:
...
     Value: 7.3.0
 Resources:
+  
   ControlPlaneNodeAutoScalingGroup:
     Type: AWS::AutoScaling::AutoScalingGroup
     DependsOn: EtcdVolume
     Properties:
       VPCZoneIdentifier:
-        - subnet-id-eu-central-1b
+        - subnet-id-eu-central-1a
```
Comment (calvix) [inline service/controller/resource/tccpn/testdata/case-0-basic-test-with-encrypter-backend-KMS-route53-enabled.golden:15]: this was caused by adding more AZs to the CP, as before there were only `eu-central-1b` and now we have `a,b,c`

**Comment by xh3b4sd** [comment]
Note that I had to do pretty much the same in https://github.com/giantswarm/aws-operator/pull/2368. I think I got it sorted. Just need to do some testing. Let's talk on Monday. We can maybe just close this one here. On the upside your PR here helped me figure out the record sets and all the templating details. That was really useful.

**Comment by calvix** [comment]
oh

**Pull request closed without merging**